### PR TITLE
Remove esbuild

### PR DESCRIPTION
**Pull request by gmalette**
Once we're satisfied that `swc` works as expected, there's no reason anymore to keep two compiler backends around. 

We can then remove support for `esbuild` entirely. 

This PR is based on top of #29 which needs to be merged first.

### CODE REVIEW

File: src/SwcCompiler.ts [hunk 1]
```diff
@@ -207,7 +207,7 @@ export class SwcCompiler implements Compiler {
 
   /** The list of globby patterns to use when searching for files to build */
   private fileGlobPatterns(config: ProjectConfig) {
-    const extensions = config.esbuild?.resolveExtensions || DefaultExtensions;
+    const extensions = config.extensions || DefaultExtensions;
```
Comment (gmalette) [inline src/SwcCompiler.ts:210]: This changes things a little. We already have a `config.extension` key, but the `config.esbuild` object also allowed defining extensions to scan. Is this change ok?

File: Readme.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ A reloading dev server for server side TypeScript projects. Compiles TypeScript
 
 ## Features
 
-- Builds and runs TypeScript really fast (using [`esbuild`](https://github.com/evanw/esbuild))
+- Builds and runs TypeScript really fast (using [`swc`](https://swc.rc))
```
Comment (gmalette) [inline Readme.md:7]: In this file I basically just `sed` `esbuild` for `swc` and same for the websites.
I think the readme could eventually benefit from some love, like configuration documentation, etc, but this PR isn't the time for that

**Comment by airhorns** [comment]
I think this is the right move long term, but, I have encountered so much pain switching between these different tools that I'd like to hold off on closing the door to esbuild for a little while longer lest we find some major blocker. Mind if we just leave these PRs open, or close and reopen in a couple months time? I appreciate you cleaning up after the changes as well!

**Comment by gmalette** [comment]
> Mind if we just leave these PRs open, or close and reopen in a couple months time

Fine by me! Let me know when you'd like to do that. 

I merged the `swc-config` branch, but there's a perf regression in it that has been fixed in #34, so ideally the next published version includes all those changes.

**Comment by airhorns** [comment]
Gonna close this for now until we decide to :fire: esbuild completely

**Pull request closed without merging**